cmd/snap/cmd_find.go

```diff
@@ -183,13 +198,20 @@ func (x *cmdFind) Execute(args []string) error {
 	cli := Client()
 
 	if x.Section != "" && x.Section != "featured" {
-		sections, err := getSections()
+		sections, err := cachedSections()
 		if err != nil {
 			return err
 		}
 		if !strutil.ListContains(sections, string(x.Section)) {
-			// TRANSLATORS: the %q is the (quoted) name of the section the user entered
-			return fmt.Errorf(i18n.G("No matching section %q, use --section to list existing sections"), x.Section)
+			// try the store just in case it was added in the last 24 hours
+			sections, err = cli.Sections()
+			if err != nil {
```

Comment:
Just a note, if the cached file is not there, and the section user requested is not found, we'll poke the store twice. Probably not a big problem anyway as the chances of this happening are rather low.

Reply:
I ... don't think so? that's why I split `cachedSections` out of `getSections`, after all.

Reply:
RIght! Got confused with the diff, sorry for marking noise.